File: Lab2/app/src/main/java/com/example/lab2/fragment/InputFragment.java
package com.example.lab2.fragment;

import android.content.Context;
import android.graphics.Typeface;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.RadioGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import com.example.lab2.ETextFont;
import com.example.lab2.R;

public class InputFragment extends Fragment {

    private OnInputFormSubmit mListener;

    public interface OnInputFormSubmit {
        void onInputFormSubmit(String text, String font);
    }

    public InputFragment() {

    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        try {
            mListener = (OnInputFormSubmit) context;
        } catch (ClassCastException e) {
            throw new ClassCastException(context + "must be implement OnInputFormSubmit");
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_input, container, false);

        EditText editText = view.findViewById(R.id.editText);
        RadioGroup radioGroup = view.findViewById(R.id.options);
        Button okButton = view.findViewById(R.id.button_ok);
        Button cancelButton = view.findViewById(R.id.button_cancel);

        okButton.setOnClickListener(v -> {
            String enteredText = editText.getText().toString();
            int selectedOptionId = radioGroup.getCheckedRadioButtonId();

            boolean isEnteredTextEmpty = enteredText.equals("");
            boolean isFontOptionChosen = selectedOptionId != -1;

            if (!isEnteredTextEmpty && isFontOptionChosen) {
                RadioButton radioButton = view.findViewById(selectedOptionId);
                Typeface typeface = radioButton.getTypeface();
                ETextFont eTextFont = ETextFont.getTextFontByTypeface(typeface);
                if (eTextFont != null) {
                    String font = ETextFont.getTextFontByTypeface(typeface).getName();
                    mListener.onInputFormSubmit(enteredText, font);
                }
            } else {
                new AlertFragment().show(getChildFragmentManager(), AlertFragment.TAG);
            }
        });

        cancelButton.setOnClickListener(v -> mListener.onInputFormSubmit("", "monospace"));

        return view;
    }

}

File: Lab3/app/src/main/java/com/example/lab3/fragment/InputFragment.kt
package com.example.lab3.fragment

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import android.widget.RadioButton
import android.widget.RadioGroup
import androidx.fragment.app.Fragment
import com.example.lab3.enumerable.ETextFont
import com.example.lab3.R

class InputFragment : Fragment() {

    private lateinit var mListener: FormSubmitListener;

    interface FormSubmitListener {
        fun onFormSubmit(text: String, font: String)
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        try {
            mListener = context as FormSubmitListener
        } catch (e: ClassCastException) {
            throw ClassCastException("$context must implement OnInputFormSubmit")
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_input, container, false);

        val okButton = view.findViewById<Button>(R.id.button_ok)
        val cancelButton = view.findViewById<Button>(R.id.button_cancel)
        val editText = view.findViewById<EditText>(R.id.edit_text)
        val radioGroup = view.findViewById<RadioGroup>(R.id.options)

        okButton.setOnClickListener {
            val enteredText = editText.text.toString()
            val selectedOptionId = radioGroup.checkedRadioButtonId
            val isEnteredTextEmpty = enteredText == ""
            val isFontOptionChosen = selectedOptionId != -1
            if (!isEnteredTextEmpty && isFontOptionChosen) {
                val radioButton = view.findViewById<RadioButton>(selectedOptionId)
                val typeface = radioButton.typeface
                val font: String = ETextFont.getTextFontByTypeface(typeface).fontName
                mListener.onFormSubmit(enteredText, font)
            } else {
                AlertFragment().show(childFragmentManager, AlertFragment.TAG)
            }
        }

        cancelButton.setOnClickListener {
            mListener.onFormSubmit(
                "",
                "monospace"
            )
        }

        return view
    }
}